### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by jkowalleck**
# enhanced base crawler & fixed URL-handling …

## nichtparasoup:
* URL-escape added where needed

## base crawler:
* increased request_timeout from 2 to 10 - some servers tend to be slow
* renamed some internal vars and functions
* added function to find html-base
* added url to return of fetchRemote function
* added base to return value of fetchHTML function
* added param for bs4features to fetchHTML function
* fixed html-meta-refresh-finder
* added abstract functions to `__init__` to intend implementation need

## crawler implementations:
* fixed location from guessing to fact-base-calculation ;-)
* fetched base from fetchHTML and used it to calc locations correctly

## import:
* reviewed py2/3 imports like described in the [Python 3 Porting Guide]
* fixed import Exception catch 
* harmonized imports
* explicit imports 
* strip unnecessary imports 

[Python 3 Porting Guide]: http://docs.pythonsprints.com/python3_porting/py-porting.html

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,9 @@
+# the webserver toolkit
 Werkzeug
+
+# basic HTML/XML toolkit
 beautifulsoup4
+
+# html/xml parser & tree builders
```
Comment (jkowalleck) [inline requirements.txt]: currently optional but i tend to use them in the near future ...

**Comment by jkowalleck** [comment]
@noqqe  review me ;-)

**Comment by noqqe** [comment]
works...

**Comment by noqqe** [comment]
nice work @jkowalleck ! :+1:

File: crawler/soupio.py [hunk 2]
```diff
@@ -11,27 +11,25 @@
...
 
-    ___uri = ""
+    __uri = ""
     __next = ""
 
-    __base = "http://soup.io/"
-
     @staticmethod
     def __build_uri(uri):
         return urlparse.urljoin(uri, "?type=image")
```
Comment (jkowalleck) [inline crawler/soupio.py]: this line was not changed - this caused #84

**Pull request merged**